comp/core/config/config.go

```diff
@@ -0,0 +1,169 @@
+	warnings *config.Warnings
+}
+
+type dependencies struct {
+	fx.In
+
+	Params internal.BundleParams
+}
+
+func newConfig(deps dependencies) (Component, error) {
+	if fxutil.IsTest && !isConfigTest {
+		panic("do not use the real comp/core/config component in tests")
+	}
+
+	warnings, err := setupConfig(
+		deps.Params.ConfFilePath,
+		deps.Params.ConfigName,
+		!deps.Params.ConfigLoadSecrets,
+		!deps.Params.ConfigMissingOK)
+	if err != nil {
+		return nil, err
+	}
+
+	if deps.Params.ConfigLoadSysProbe {
+		_, err := sysconfig.Merge(deps.Params.SysProbeConfFilePath)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return &cfg{warnings}, nil
+}
+
+func (c *cfg) IsSet(key string) bool {
+	return config.Datadog.IsSet(key)
+}
+func (c *cfg) Get(key string) interface{} {
+	return config.Datadog.Get(key)
+}
+func (c *cfg) GetString(key string) string {
+	return config.Datadog.GetString(key)
+}
+func (c *cfg) GetBool(key string) bool {
+	return config.Datadog.GetBool(key)
+}
+func (c *cfg) GetInt(key string) int {
+	return config.Datadog.GetInt(key)
+}
+func (c *cfg) GetInt32(key string) int32 {
+	return config.Datadog.GetInt32(key)
+}
+func (c *cfg) GetInt64(key string) int64 {
+	return config.Datadog.GetInt64(key)
+}
+func (c *cfg) GetFloat64(key string) float64 {
+	return config.Datadog.GetFloat64(key)
+}
+func (c *cfg) GetTime(key string) time.Time {
+	return config.Datadog.GetTime(key)
+}
+func (c *cfg) GetDuration(key string) time.Duration {
+	return config.Datadog.GetDuration(key)
+}
+func (c *cfg) GetStringSlice(key string) []string {
+	return config.Datadog.GetStringSlice(key)
+}
+func (c *cfg) GetFloat64SliceE(key string) ([]float64, error) {
+	return config.Datadog.GetFloat64SliceE(key)
+}
+func (c *cfg) GetStringMap(key string) map[string]interface{} {
+	return config.Datadog.GetStringMap(key)
+}
+func (c *cfg) GetStringMapString(key string) map[string]string {
+	return config.Datadog.GetStringMapString(key)
+}
+func (c *cfg) GetStringMapStringSlice(key string) map[string][]string {
+	return config.Datadog.GetStringMapStringSlice(key)
+}
+func (c *cfg) GetSizeInBytes(key string) uint {
+	return config.Datadog.GetSizeInBytes(key)
+}
+func (c *cfg) AllSettings() map[string]interface{} {
+	return config.Datadog.AllSettings()
+}
+func (c *cfg) AllSettingsWithoutDefault() map[string]interface{} {
+	return config.Datadog.AllSettingsWithoutDefault()
+}
+func (c *cfg) AllKeys() []string {
+	return config.Datadog.AllKeys()
+}
+func (c *cfg) GetKnownKeys() map[string]interface{} {
+	return config.Datadog.GetKnownKeys()
+}
+func (c *cfg) GetEnvVars() []string {
+	return config.Datadog.GetEnvVars()
+}
+func (c *cfg) IsSectionSet(section string) bool {
+	return config.Datadog.IsSectionSet(section)
+}
+func (c *cfg) Warnings() *config.Warnings {
+	return c.warnings
+}
+
+func newMock(deps dependencies, t testing.TB) Component {
+	old := config.Datadog
+	config.Datadog = config.NewConfig("mock", "XXXX", strings.NewReplacer())
```

Comment:
At the moment, this is wrapping pkg/config so this means that anything accessing config via pkg/config will also get the mocked values.  When we aren't using pkg/config anymore, this will be gone.

Note that this is the same thing that config.NewMock does.

Reply:
This is unusual to set a global variable inside a mock. 
Is the idea to use a Mock Component when writing new unit tests so that when the real code handles `Config.Component`, unit tests don't need to be updated?

Reply:
Yep, all use of the global `config.Datadog` will _eventually_ go away, but for the moment we need to keep it in sync with the component.